src/main/Core/Dock/DockModule.ts

```diff
@@ -1,7 +1,22 @@
-import type { App } from "electron";
+import type { Dependencies } from "@Core/Dependencies";
+import type { DependencyRegistry } from "@Core/DependencyRegistry";
 
 export class DockModule {
-    public static bootstrap(app: App) {
-        app.dock?.hide();
+    public static bootstrap(dependencyRegistry: DependencyRegistry<Dependencies>) {
+        const app = dependencyRegistry.get("App");
+        const settingsManager = dependencyRegistry.get("SettingsManager");
+        const eventSubscriber = dependencyRegistry.get("EventSubscriber");
+
+        const setAppIconVisibility = () => {
+            const showAppIconInDock = settingsManager.getValue("appearance.showAppIconInDock", false);
+            if (showAppIconInDock) {
+                app.dock?.show();
+            } else {
+                app.dock?.hide();
+            }
```

Comment:
There is no reason to have this variable as it's just used once and `"appearance.showAppIconInDock"` is also pretty clear what value represents:

```suggestion
            if (settingsManager.getValue("appearance.showAppIconInDock", false)) {
                app.dock?.show();
            } else {
                app.dock?.hide();
            }
```